src/Temporalio/Common/Priority.cs

```diff
@@ -0,0 +1,60 @@
+namespace Temporalio.Common
+{
+    /// <summary>
+    /// Priority contains metadata that controls relative ordering of task processing when tasks are
+    /// backlogged in a queue. Initially, Priority will be used in activity and workflow task
+    /// queues, which are typically where backlogs exist. Priority is (for now) attached to
+    /// workflows and activities. Activities and child workflows inherit Priority from the workflow
+    /// that created them, but may override fields when they are started or modified. For each field
+    /// of a Priority on an activity/workflow, not present or equal to zero/empty string means to
+    /// inherit the value from the calling workflow, or if there is no calling workflow, then use
+    /// the default (documented on the field).
+    ///
+    /// The overall semantics of Priority are:
+    /// 1. First, consider "priority_key": lower number goes first.
+    /// (more will be added here later).
+    /// </summary>
+    public sealed record Priority
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Priority"/> class.
+        /// </summary>
+        /// <param name="priorityKey">The priority key.</param>
+        public Priority(int? priorityKey = null) => PriorityKey = priorityKey;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Priority"/> class from a proto.
+        /// </summary>
+        /// <param name="priority">The proto to initialize from.</param>
+        internal Priority(Temporalio.Api.Common.V1.Priority priority)
+            : this(priority?.PriorityKey != 0 ? priority?.PriorityKey : null)
+        {
+        }
+
+        /// <summary>
+        /// Gets the default Priority instance.
+        /// </summary>
+        public static Priority Default => new();
```

Comment:
```suggestion
        public static Priority Default { get; } = new();
```

Forgot to mention this on other PR. We shouldn't reinstantiate this every time the property is called.